test/functional/wallet_decommission_request.py

```diff
@@ -0,0 +1,323 @@
+            self.log.debug(f'Tip: {tip_id}')
+
+            # Submit a valid transaction
+            output = {
+                    'Transfer': [ { 'Coin': 50_000 * ATOMS_PER_COIN }, { 'PublicKey': {'key': {'Secp256k1Schnorr' : {'pubkey_data': pub_key_bytes}}} } ],
+            }
+            encoded_tx, tx_id = make_tx([reward_input(tip_id)], [output], 0)
+            self.log.debug(f"Encoded transaction {tx_id}: {encoded_tx}")
+
+
+            self.setup_pool_and_transfer([encoded_tx])
+
+            # sync the wallet
+            assert_in("Success", await wallet.sync())
+
+            # check wallet best block if it is synced
+            best_block_height = await wallet.get_best_block_height()
+            assert_in(best_block_height, '1')
+
+            balance = await wallet.get_balance()
+            assert_in("Coins amount: 50000", balance)
+
+            # create new account that would hold decommission key
+            assert_in("Success", await wallet.create_new_account())
+            assert_in("Success", await wallet.select_account(1))
+            acc1_address = await wallet.new_address()
+            decommission_pub_key_bytes = await wallet.new_public_key()
+            decommission_pub_key_encoded  = pub_key_obj.encode(self.public_key(decommission_pub_key_bytes.hex())).to_hex()[2:]
+
+            assert_in("Success", await wallet.select_account(DEFAULT_ACCOUNT_INDEX))
+            assert_in("The transaction was submitted successfully", await wallet.send_to_address(acc1_address, 100))
+            assert_in("Success", await wallet.select_account(1))
+            transactions = node.mempool_transactions()
+
+            assert_in("Success", await wallet.select_account(DEFAULT_ACCOUNT_INDEX))
+
+            assert_in("The transaction was submitted successfully", await wallet.create_stake_pool(40000, 0, 0.5, decommission_pub_key_encoded))
+            transactions2 = node.mempool_transactions()
+            for tx in transactions2:
+                if tx not in transactions:
+                    transactions.append(tx)
+
+            self.gen_pos_block(transactions, 2)
+            assert_in("Success", await wallet.sync())
+
+            pools = await wallet.list_pool_ids()
+            assert_equal(len(pools), 1)
+            assert_equal(pools[0].balance, 40000)
+
+            # create decommission request
+            decommission_req_output = await wallet.decommission_stake_pool_request(pools[0].pool_id)
+            decommission_req = decommission_req_output.split('\n')[1]
+
+            # switch accounts and sign decommission request
+            assert_in("Success", await wallet.select_account(1))
+            decommission_signed_tx_output = await wallet.sign_raw_transaction(decommission_req)
+            decommission_signed_tx = decommission_signed_tx_output.split('\n')[1]
+            assert_in("The transaction was submitted successfully", await wallet.submit_transaction(decommission_signed_tx))
+
```

Comment:
We need a test that creates the decommission request, signs it in another wallet, and submits it in the first wallet.

Reply:
as the test is using 2 accounts it should be similar to using 2 wallets, as each account has access to different private keys.

Would be good to add an assert that trying to sign the decommission request transaction with the other account will fail.

Reply:
the problem is that I'm not sure if python test framework allows to run several wallets. And what problems should be solved to do that. So I decided to write tests using accounts which proves the point but is much faster to do.

Reply:
I can live with one wallet for the reasons Boris explained... but more robust testing is needed to test the counter case too, and achieve the fully signed transaction gradually, and ensure it works.

Reply:
If you want to create 2 wallets you can do it with just adding
       ` async with WalletCliController(node, self.config, self.log) as wallet, WalletCliController(node, self.config, self.log) as wallet2:`
       
With just 1 change to the `create_wallet(self)` function needs to get a wallet_file as currently it always creates it with the same name, so that both wallets can have different files, and it will work fine